File: templates/Frameworks/MVVMLight/Project/Activation/DefaultLaunchActivationHandler_postaction.cs
class DefaultLaunchActivationHandler : ActivationHandler<LaunchActivatedEventArgs>
{
    //{[{
    private readonly string _navElement;

    private NavigationServiceExt NavigationService
    {
        get
        {
            return Microsoft.Practices.ServiceLocation.ServiceLocator.Current.GetInstance<NavigationServiceExt>();
        }
    }

    public DefaultLaunchActivationHandler(Type navElement)
    {
        _navElement = navElement.FullName;
    }

    //}]}
}

File: templates/_composition/MVVMLight/ProjectVB/Activation/DefaultLaunchActivationHandler_postaction.vb
Namespace Activation
    Friend Class DefaultLaunchActivationHandler
        Inherits ActivationHandler(Of LaunchActivatedEventArgs)
'{[{

        Private ReadOnly _navElement As string

        Private ReadOnly Property NavigationService As NavigationServiceEx
            Get
                Return Microsoft.Practices.ServiceLocation.ServiceLocator.Current.GetInstance(Of NavigationServiceEx)()
            End Get
        End Property

        Public Sub New(navElement As Type)
            _navElement = navElement.FullName
        End Sub
'}]}
    End Class
End Namespace
